src/app/view/editor-main-view/data-views/trainrunSectionViewObject.ts

```diff
@@ -156,6 +156,17 @@ export class TrainrunSectionViewObject {
     d.getPath().forEach((p) => {
       key += p.toString();
     });
+
+    // Include source and target node positions to detect when they move
+    key += "_SRC_" + d.getSourceNode().getPositionX() + "_" + d.getSourceNode().getPositionY();
+    key += "_TRG_" + d.getTargetNode().getPositionX() + "_" + d.getTargetNode().getPositionY();
```

Comment:
This would not be required if we properly set the key entries above, by replacing `d.getPath()` with `getCollapsedChainPath()`?

Reply:
Fixed in #605